# Homework 6 (Due Tue. 3/20 before class)

**Name:**

**Learning Objectives Assessed:**

- Formulating linear programs (LP) to model an optimization problem. (Model)
- Solving LPs of two variables manually by graphing. (Analyze)
- Solving LPs using Gurobi in Python. (Code)
- Interpreting shadow price information. (Analyze)
- Communicate LP formulations using LaTex. (Communicate)

**For this homework, you must add your response to this Jupyter notebook and submit it.**

## Instructions for typesetting LP formulations in Markdown cells
In questions 1,3,4,5, you are asked to "formulate a LP." This means to define your decision variables, then write the objective function and constraints. An example of a LP formulation is given below (this is the LP from the class on Thursday 3/1).

Let decision variables $X$ and $Y$ denote the amount of product X and product Y to use respective. The linear program is
$$\begin{aligned}
\text{maximize} && 20X+10Y \\
\text{subject to:} \\
\text{Material 1:} && 4X+Y & \le 60 \\
\text{Material 2:} && 2Y & \le 48 \\
\text{Labor:} && X+Y & \le 30 \\
\text{Non-negativity} && X,Y & \ge 0
\end{aligned}$$

Notice that the variables $X$ and $Y$ are in a special font denoting mathematical variables. Moreover, notice that the linear program above is centered and aligned, both at the colons and at the signs ($\le$ and $\ge$). It is useful to learn to do such formating in Markdown in order to better communicate your LP formulations to others.

Double click this Markdown cell to see the code that created the math. Then execute the cell again to render it.

As you can see, to render a mathematical expression nicely, we enclose it using dollar signs. So the expression X>0, becomes $X>0$. A single dollar sign is for mathematical expressions inline, and double dollar signs are for a standalone mathematical expressions in its own line (See code by double clicking this cell). $$X>=0$$

To make the linear program aligned, we not only use the double dollar signs, but also use the `\begin{aligned} \end{aligned}` commands. (Double click the linear program above to see the code.) Within the alignment block:

- `\text{ }` is for displaying the enclosed string as plain text, without the mathematical rendering.
- `&` is for alignment. The convention is right align before the first `&`, then left align between the first and second `&` of each line, then right align again between the second and third `&` and so on. Hence, to make it right aligned both before and after the colon, we use a double `&&` after the `\text{}`. If this is confusing, you can simply copy this convention for this homework (`&&` after the explanation of constraint, and `&` before the sign). 
- `\\` is for creating a new line. Notice that we end the line early using `\\` for "subject to" and "maximize".
- `\le` (less than or equal to) is for $\le$, and `\ge` (greater than or equal to) is for $\ge$. This looks better than $<=$ and $>=$.

This notation is called LaTex, and is the standard for people in technical fields to display math. Make sure that you close any braces you begin and put the alignment characters `&` and newlines `\\` at the right places, or the math may not render. You can see another example in the prompt to Q2 (double click the prompt of Q2 to see the LaTex code).

## Q1 (Solving a LP manually by graphing)

Solve DMD Exercise 7.2. The question begins with "The Magnetron Company manufactures and markets microwave ovens." For part a), you should type your formulation into a Markdown cell following the typesetting instructions above. For parts b) and c), you should plot by hand on a scrap piece of paper, and type the answer in a Markdown cell. **You do not have to submit your hand drawing**, but you need to learn to do this for a quiz or an exam. 

Let decision variables $F$ and $C$ be the number of Full Size and Compact Ovens manufactured by Magnetron Company. The LP formulation is given as below:
$$\begin{aligned}
\text{maximize} && 120F+130C \\
\text{subject to:} \\
\text{General Assemby Hours:} && 2F+1C & \le 500 \\
\text{Electronic Assembly Hours:} && 2F+3C & \le 800 \\
\text{Demand for Full Size Ovens:} && F & \le 220 \\
\text{Demand for Compact Ovens:} && C & \le 180 \\
\text{Non-Negativity:} && F,C & \ge 0
\end{aligned}$$

## The optimal solution is provided below:
$$\begin{aligned}
\text{Full Size Ovens:} && 175\\
\text{Compact Ovens:} && 150\\
\text{Optimal Profit:} && 40500
\end{aligned}$$

## Q2 (Solving a LP using Gurobi)

**a)** Solve the LP for the NBS supply problem (DMD section 7.1) using Gurobi. Print the optimal solution, as well as the shadow price of each constraint. For your convenience, the LP formulation is reproduced below. 

Let decision variables $A$, $B$, $C$, $D$, $E$, $F$, $G$, and $H$ correspond to the amount of coal to be contracted from Ashley, Bedford, Consol, Dunby, Earlam, Florence, Gaston and Hopt respectively (in mtons).

$$\begin{aligned}
 \text{minimize} && 49.5A+50B+61C+63.5D+66.5E+71F+72.5G+80H \\
 \text{subject to:} && \\
 \text{Supply:} && A+B+C+D+E+F+G+H & = 1225 \\
 \text{Union:} && A+B-C+D-E+F-G-H & \ge 0 \\
 \text{Truck:}&& B+D+E+F & \le 720 \\
 \text{Rail:}&& A+C+G+H & \le 650 \\
 \text{Volatility:} && -4A-3B-C+D+2E+3F+4G+6H & \ge 0 \\
 \text{Acap:}&& A &\le 300 \\
\text{Bcap:}&& B &\le 600 \\
\text{Ccap:}&& C &\le 510 \\
\text{Dcap:}&& D &\le 655 \\
\text{Ecap:}&& E &\le 575 \\
\text{Fcap:}&& F &\le 680 \\
\text{Gcap:}&& G &\le 450 \\
\text{Hcap:}&& H &\le 490 \\
\text{Non-negativity:} && A,B,C,D,E,F,G,H & \ge 0
\end{aligned}$$


In [49]:
import gurobipy as grb
mod = grb.Model()
A = mod.addVar(lb=0)
B = mod.addVar(lb=0)
C = mod.addVar(lb=0)
D = mod.addVar(lb=0)
E = mod.addVar(lb=0)
F = mod.addVar(lb=0)
G = mod.addVar(lb=0)
H = mod.addVar(lb=0)

mod.setObjective(49.5*A+50*B+61*C+63.5*D+66.5*E+71*F+72.5*G+80*H,sense = grb.GRB.MINIMIZE)
s = mod.addConstr(A+B+C+D+E+F+G+H >= 1225) ##need to ask professor
u = mod.addConstr(A+B-C+D-E+F-G-H>=0)
t = mod.addConstr(B+D+E+F<=720)
r = mod.addConstr(A+C+G+H<=650)
v = mod.addConstr(-4*A-3*B-C+D+2*E+3*F+4*G+6*H>=0)
acap = mod.addConstr(A<=300)
bcap = mod.addConstr(B<=600)
ccap = mod.addConstr(C<=510)
dcap = mod.addConstr(D<=655)
ecap = mod.addConstr(E<=575)
fcap = mod.addConstr(F<=680)
gcap = mod.addConstr(G<=450)
hcap = mod.addConstr(H<=490)
##do not print anything when calling mod.optimize()
mod.setParam('OutputFlag',False)
mod.optimize()
print('Optimal Objective: {0:.2f}'.format(mod.ObjVal))
print('Optimal Solution:')
print('\tA = {0:.1f}'.format(A.x))
print('\tB = {0:.1f}'.format(B.x))
print('\tC = {0:.1f}'.format(C.x))
print('\tD = {0:.1f}'.format(D.x))
print('\tE = {0:.1f}'.format(E.x))
print('\tF = {0:.1f}'.format(F.x))
print('\tG = {0:.1f}'.format(G.x))
print('\tH = {0:.1f}'.format(H.x))
print('Shadow Prices:')
print('\tSupply Constraint: {0:.2f}'.format(s.PI))
print('\tUnion Constraint: {0:.2f}'.format(u.PI))
print('\tTruck Capacity Constraint: {0:.2f}'.format(t.PI))
print('\tRail Capacity Constraint: {0:.2f}'.format(r.PI))
print('\tVolatility Constraint: {0:.2f}'.format(v.PI))
print('\tAcap: {0:.2f}'.format(acap.PI))
print('\tBcap: {0:.2f}'.format(bcap.PI))
print('\tCcap: {0:.2f}'.format(ccap.PI))
print('\tDcap: {0:.2f}'.format(dcap.PI))
print('\tEcap: {0:.2f}'.format(ecap.PI))
print('\tFcap: {0:.2f}'.format(fcap.PI))
print('\tGcap: {0:.2f}'.format(gcap.PI))
print('\tHcap: {0:.2f}'.format(hcap.PI))

Optimal Objective: 73267.50
Optimal Solution:
	A = 55.0
	B = 600.0
	C = 0.0
	D = 20.0
	E = 100.0
	F = 0.0
	G = 450.0
	H = 0.0
Shadow Prices:
	Supply Constraint: 61.50
	Union Constraint: 0.00
	Truck Capacity Constraint: -1.00
	Rail Capacity Constraint: 0.00
	Volatility Constraint: 3.00
	Acap: 0.00
	Bcap: -1.50
	Ccap: 0.00
	Dcap: 0.00
	Ecap: 0.00
	Fcap: 0.00
	Gcap: -1.00
	Hcap: 0.00


## The supply constraint has a shadow price of 61.50 which implies that if New Bedford decides to source extra coking coal then the maximum cost of such coking coal that the company should be willing to pay is dollars 61.5/mton
## Union constraint has a shadow price of 0 which implies that sourcing more coal from union mines would not result in any change in the optimal cost of procurement
## Truck capacity constraint has a shadow price of -1.00 which implies that increasing the truck transport capacity by 1 mton shall reduce the optimal cost of procurement by dollars 1 per mton of truck capacity increase
## Rail capacity constraint has a shadow price of 0 which implies that increasing the capacity of rail transport shall not result in any change in the optimal cost of procurement
## Capacity contraint of Ashley has a shadow price of 0 which implies that increasing the capacity at Ashley shall not result in any change in the optimal cost of procurement. In other words, Ashley capacity is not a binding constraint. Similarly, capacities at Consol, Dunby, Earlam, Florence and Hopt are not binding contraints
## Capacity at Bedford mining has a shadow price of -1.50 which implies that increasing the capacity at Bedford shall result in reduction of dollars 1.50 per mton of capacity increase in the optimal cost of procurement
## Similarly, for Gaston, every 1 mton increase in mining capacity at Gaston shall result in reduction of  dollars 1 in the optimal cost of procurement

In [61]:
# Example output (using print statements after solving the model)

Optimal solution
	A=55.0
	B=600.0
	C=0.0
	D=20.0
	E=100.0
	F=0.0
	G=450.0
	H=0.0

Shadow prices
	Supply constraint: 61.5
	Union constraint: 0.0
	Truck constraint: -1.0
	Rail constraint: 0.0
	Volatility constraint: 3.0
	Acap: 0.0
	Bcap: -1.5
	Ccap: 0.0
	Dcap: 0.0
	Ecap: 0.0
	Fcap: 0.0
	Gcap: -1.0
	Hcap: 0.0


**b)** For each of the constraints except for the volatility constraint, write 1-2 sentence describing what insights can be obtained from the shadow price. You may check your answers using DMD section 7.5.4, but please write the sentence in your own words. For the capacity constraints with zero shadow cost (Acap, Ccap, Dcap, Ecap, Fcap, Hcap), you only need to describe for one constraint, and say that the others are like it.

Example sentence: The shadow price of the union constraint is zero, which means that the union is not a bottleneck in this scenario. (In other words, NBS Corporation would not be able to benefit from the union relaxing its requirement on coal contracts, by allowing for slightly more coal to be contracted from non-union suppliers than from union suppliers.)

## Q3 (Production Planning)

Solve DMD Exercise 7.8. The question begins with "Nature's best Frozen Foods company produces four different mixes of frozen ready-to-eat vegetables. 

For part a), you should formulate the linear program following the instructions at the top of this homework (define the decision variables then write the LP nicely in a Markdown cell using LaTex).

## DMD 7.8 Let decision variables S,B,M,V denote the total packets of Stir Fry, Barbeque, Hearty Mushrooms and Veggie Crunch packets to be produced by Nature's Best Frozen Foods respectively. The LP formulation is given as below:
$$\begin{aligned}
\text{maximize} && 0.22S+0.20B+0.18M+0.18V \\
\text{subject to:} \\
\text{Carrot Constraint:} && 2.5S+2B+2.5V & \le 150000 \\
\text{Mushrooms Constraint:} && 3S+4V & \le 80000 \\
\text{Green Peppers Constraint:} && 2.5S+2B+3M+2.5V & \le 135000 \\
\text{Brocolli Consraint:} && 2S+3B+3M+2.5V & \le 140000 \\
\text{Corn Consraint:} && 3B+2.5V & \le 150000 \\
\text{Non-Negativity:} && S,B,M,V & \ge 0
\end{aligned}$$

In [96]:
mod = grb.Model()
S = mod.addVar(lb=0)
B = mod.addVar(lb=0)
M = mod.addVar(lb=0)
V = mod.addVar(lb=0)

mod.setObjective(0.22*S+0.20*B+0.18*M+0.18*V,sense = grb.GRB.MAXIMIZE)

CA = mod.addConstr(2.5*S+2*B+2.5*V<=150000)
MU = mod.addConstr(3*S+4*V<=80000)
GP = mod.addConstr(2.5*S+2*B+3*M+2.5*V<=135000)
BR = mod.addConstr(2*S+3*B+3*M+2.5*V<=140000)
CO = mod.addConstr(3*B+2.5*V<=150000)

mod.setParam('OutputFlag',False)
mod.optimize()

print('Optimal Objective Value:{0:.2f}'.format(mod.ObjVal))
print('Optimal Solution:')
print('\tStir Fry:{0:.0f}'.format(S.x))
print('\tBarbeque:{0:.0f}'.format(B.x))
print('\tHearty Mushroom:{0:.1f}'.format(M.x))
print('\tVeggie Crunch:{0:.1f}'.format(V.x))
print('Shadow Price for Carrots: {0:.2f}'.format(CA.PI))
print('Shadow Price for Mushrooms: {0:.2f}'.format(MU.PI))
print('Shadow Price for Green Peppers: {0:.2f}'.format(GP.PI))
print('Shadow Price for Brocolli: {0:.2f}'.format(BR.PI))
print('Shadow Price for Corn: {0:.2f}'.format(CO.PI))

Optimal Objective Value:11644.44
Optimal Solution:
	Stir Fry:26667
	Barbeque:28889
	Hearty Mushroom:0.0
	Veggie Crunch:0.0
Shadow Price for Carrots: 0.00
Shadow Price for Mushrooms: 0.03
Shadow Price for Green Peppers: 0.00
Shadow Price for Brocolli: 0.07
Shadow Price for Corn: 0.00


## c) As may be seen from above the shadow price of Green Pepper contraint is 0 which implies that increasing the supply for Green Pepper will not affect the optimal solution and thus we shouldnt increase the supply for Green Peppers. However, we can increase the supply for Mushrooms and Brocolli which shall result in increase in the profit by dollar 0.03 and 0.07 per oz. increase in supply respectively.

## Q4 (Formulating Constraints)

Solve DMD Exercise 7.12. Before solving using Gurobi, you should formulate the linear program by defining the decision variables and write the objective and constraints nicely using LaTex.


## 7.12 Let decision variables A, B, C, D and E denote the fraction of total investment, invested in Adams, Barney, Chilton, Dunster and Excelsior respectively. The LP formulation is given as below:
$$\begin{aligned}
\text{maximize} && 4.5A+5.62B+6.8C+10.15D+20.60E \\
\text{subject to:} \\
\text{Total Investment Constraint:} && A+B+C+D+E & \le 1.00 \\
\text{Money Market Constraint:} && A+B & \le 0.3 \\
\text{Aggressive Growth Fund Constraint:} && E & \le 0.2 \\
\text{Average Risk Consraint:} && A+2B+2C+3D+5E & \le 2.5 \\
\text{Non-Negativity:} && A,B,C,D,E & \ge 0
\end{aligned}$$

In [94]:
mod = grb.Model()
A = mod.addVar(lb=0)
B = mod.addVar(lb=0)
C = mod.addVar(lb=0)
D = mod.addVar(lb=0)
E = mod.addVar(lb=0)

#mod.setObjective(0.045*A+0.0562*B+0.068*C+0.1015*D+0.2060*E,sense = grb.GRB.MAXIMIZE)
mod.setObjective(4.5*A+5.62*B+6.8*C+10.15*D+20.60*E,sense = grb.GRB.MAXIMIZE)
#total contraint
mod.addConstr(A+B+C+D+E<=1)

MA = mod.addConstr(A+B<=0.30)
GF = mod.addConstr(E<=0.20)
RI = mod.addConstr(A+2*B+2*C+3*D+5*E<=2.50)


mod.setParam('OutputFlag',False)
mod.optimize()

print('Optimal Objective Value:{0:.2f}%'.format(mod.ObjVal))
print('Optimal Solution:')
print('\tAdams:{0:.3f}'.format(A.x))
print('\tBarney:{0:.3f}'.format(B.x))
print('\tChilton:{0:.3f}'.format(C.x))
print('\tDunster:{0:.3f}'.format(D.x))
print('\tExcelsior:{0:.3f}'.format(E.x))
print('Shadow Price for Money Market Funds: {0:.2f}'.format(MA.PI))
print('Shadow Price for Aggressive Growth Fund: {0:.2f}'.format(GF.PI))
print('Shadow Price for Average Risk: {0:.2f}'.format(RI.PI))

Optimal Objective Value:9.54
Optimal Solution:
	Adams:0.300
	Barney:0.000
	Chilton:0.300
	Dunster:0.200
	Excelsior:0.200
Shadow Price for Money Market Funds: 1.05
Shadow Price for Aggressive Growth Fund: 3.75
Shadow Price for Average Risk: 3.35


## The above solution states that the investor should allocate 30% of funds in Adams, 30% funds in Chilton, 20% funds in Dunster and 20% funds in Excelsior to maximize returns. The optimal average annual return is 9.54%

## Q5 (Production and Transportation)

Solve DMD Exercise 7.9. The question begins with "Johnson Electric produces small electric motors for four appliance manufacturers in each of the three plants). 

### Additional instructions and hints
When formulating the linear program in part a), **instead of manually defining each variable as in the hint in the book, you should practice using indexing notation**, which also saves you effort on manual input and makes your formulation easily generalizable to larger data sets. You should use the following notation in your formulation:

- $I=\{A,B,C\}$ represents the set of plants;
- $J=\{O,T,H,D\}$ represent the set of manufacturers. 
- Let your decision variables be $x_{ij}$, which is the amount shipped from plant $i \in I$ to manufacturer $j \in J$. (For example $x_{AO}$ is the amount shipped from Arlington to Onyx. $\sum_{j\in J} x_{Aj}$ is the total amount produced from Arlington. $\sum_{i \in I} x_{iO}$ is the total amount shipped to Onyx.)
- For each plant $i \in I$, the production cost is $p_i$ and the monthly capacity is $q_i$. (For example, $p_A=17$ and $q_A=800$.)
- For each manufactuer $j \in J$, the demand of motors per month is given by $d_j$. (For example, $d_O=300$, $d_T=500$.) 
- For each plant $i\in I$ and each manufacturer $j\in J$, the transportation cost from plant $i$ to manufactuer $j$ can be indexed by $c_{ij}$. (For example using the data from the table associated with the question, $c_{AO}=3$, $c_{BT}=4$.)

To write a indexed variable in Latex, you can use `c_{ij}` for $c_{ij}$ (make sure to enclose it either in single dollar sign for in-line math or double-dollar sign for stand-alone math). To write summation, you can use the command `\sum`, with underline and braces to indicate the set you are summing over. For example (double click the following expression to see the code), 
$$\text{Total transportation cost} = \sum_{i \in I, j \in J} c_{ij}x_{ij}.$$

The above expression is equivalent to looping over both the set of plants $I$ and set of manufacturers $J$ and adding the term $c_{ij} x_{ij}$. 

The summation notation helps you express many constraints at once. For example you can write the capacity constraint at all three plants using the equation
$$ \text{Respecting capacity:  } \sum_{j \in J} x_{ij} \le q_i \text{ for every plant $i \in I$}.$$

The decision variables as well as the above constraints can be inputed into Gurobi using the following code.

```python
import gurobipy as grb
import pandas as pd

I=['Arlington','Binghamton','Canton']
J=['Onyx','Treble','Hilton','Dean']

mod=grb.Model()

x={}
for i in I:
    for j in J:
        x[i,j]=mod.addVar(lb=0,name='x[{0},{1}]'.format(i,j))

q=pd.Series([800,600,700],index=I)
        
for i in I:
    mod.addConstr(sum(x[i,j] for j in J) <= q.loc[i],name='Capacity for '+i)

```
Note the use of Python command `sum` for constructing summations $\sum_{j \in J} x_{ij}$. The command `q.loc[i]` accesses the value of pandas Series `q` corresponding to $i$. For example, `q.loc['Arlington']` is 800.

**Write your formulation below using LaTex, then solve using Gurobi:**


## 7.9 Let $I = \{A,B,C\}$ be the set of plants and $J=\{O,T,H,D\}$ be the set of customers. $x_{ij}$ is the quantity of electric motors shipped from plant $i \in I$ to customer $j \in J$. $c_{ij}$ is the cost of shipping 1 electric motor from plant $i \in I$ to customer $j \in J$. 
This implies that $\sum_{j \in J} x_{Aj}$ is the total quantity produced from Plant A and $\sum_{i \in I} x_{iO}$ is the total quantity shipped to customer O.
For each plant $i \in I$, the production cost is $p_{i}$ and the total monthly production capacity is $q_{i}$.
For each customer $i \in J$ the monthly demand is $d_{j}$. For each plant $i \in I$ and each customer $j \in J$, the cost of shipping is given by $c_{ij}$.

$$\begin{aligned}
\text{minimize:} && \sum_{i \in I,j \in J} c_{ij}x_{ij} + \sum_{i \in I} p_{i} \sum_{j \in J}x_{ij} \\
\text{subject to:} \\
\text{Production Capacity Constraint:} && \sum_{j \in J} x_{ij} \le q_{i} \text { for each plant $i\in I$} \\
\text{Demand Constraint:} && \sum_{i \in I} x_{ij} \ge d_{j} \text{ for every customer $j \in J$} \\
\text{Non-Negativity Constraint:} && \sum_{i \in I,j \in J} x_{ij} \ge 0 \\
\end{aligned}$$

In [131]:
import pandas as pd

I = ['A','B','C']
J = ['O','T','H','D']

mod = grb.Model()
x={}
for i in I:
    for j in J:
        x[i,j]= mod.addVar(lb=0,name='x[{0},{1}]'.format(i,j))
q = pd.Series([800,600,700],index = I)   
p = pd.Series([17,20,24],index=I)
d = pd.Series([300,500,400,600],index=J)
c = pd.DataFrame([[3,2,5,7],[6,4,8,3],[9,1,5,4]],index=I, columns=J)
for i in I:
    mod.addConstr(sum(x[i,j] for j in J) <=q.loc[i],name = 'Capacity[{0}]'.format(i))
for j in J:
    mod.addConstr(sum(x[i,j] for i in I) >= d.loc[j],name = 'Demand [{0}]'.format(j) )
##define the objective function
#mod.setObjective(sum(sum(c.loc[i,j]*x[i,j] for i in I for j in J),sum(p.loc[i]*x[i,j] for i in I for j in J)),sense = grb.GRB.MINIMIZE)
#mod.setObjective(sum(c.loc[i,j]*x[i,j] for i in I for j in J),sense = grb.GRB.MINIMIZE)
mod.setObjective(sum(c.loc[i,j]*x[i,j] for i in I for j in J)+sum(p.loc[i]*x[i,j] for i in I for j in J),sense = grb.GRB.MINIMIZE)

#Run the optimization and set outputflag to 0 so that intermittent results are not shown
mod.setParam('OutputFlag',False)
mod.optimize()

##print out the results
print('Optimal Objective Value: {0:.2f}'.format(mod.ObjVal))
print('Optimal solution')
for var in mod.getVars():
    print('\t{0} = {1:.2f}'.format(var.VarName,var.x))
print('Shadow prices:')
for constr in mod.getConstrs():
    print('\t{0} = {1:.2f}'.format(constr.ConstrName,constr.PI))


Optimal Objective Value: 40500.00
Optimal solution
	x[A,O] = 300.00
	x[A,T] = 100.00
	x[A,H] = 400.00
	x[A,D] = 0.00
	x[B,O] = 0.00
	x[B,T] = 0.00
	x[B,H] = 0.00
	x[B,D] = 600.00
	x[C,O] = 0.00
	x[C,T] = 400.00
	x[C,H] = 0.00
	x[C,D] = 0.00
Shadow prices:
	Capacity[A] = -6.00
	Capacity[B] = -1.00
	Capacity[C] = 0.00
	Demand [O] = 26.00
	Demand [T] = 25.00
	Demand [H] = 28.00
	Demand [D] = 24.00


### b) Based on the above results, we can see that capacity at Arlington and Binghamton is being fully utilized and is a binding constraint while capacity at Canton is not utilized fully. Therefore, any further capital cost should be incurred in increasing production capacity at Arlington as there is higher reduction of 6 dollars per unit increase in capacity and then at Binghamton. 
### Increase in demand at Dean shall lead to the lowest increase in costs to the company and should be favored as against other customers
